types/tx.go

```diff
@@ -118,3 +120,18 @@ type TxResult struct {
 	Tx     Tx                     `json:"tx"`
 	Result abci.ResponseDeliverTx `json:"result"`
 }
+
+// ComputeAminoOverhead calculates the overhead for amino encoding a transaction.
+// The overhead consists of varint encoding the field number and the wire type
+// (= length-delimited = 2), and another varint encoding the length of the
+// transaction.
+// The field number can be the field number of the particular transaction, or
+// the field number of the parenting struct that contains the transactions []Tx
+// as a field (this field number is repeated for each contained Tx).
+// If some []Tx are encoded directly (without a parenting struct), the default
+// fieldNum is also 1 (see BinFieldNum in amino.MarshalBinaryBare).
+func ComputeAminoOverhead(tx Tx, fieldNum int) int64 {
```

Comment:
Protobuf is so silly :/, the type is getting prepended to every single element of the slice. Why wouldn't they just make a native concept of slices with a length prefix at the beginning :L.

Reply:
Protobuf can actually behave as you'd expect it. But only for primitive types. `bytes` doesn't seem to fall in this category (as it is length-delimited) ¯\_(ツ)_/¯ 

Otherwise, you could do sth along the lines of:
```proto
message Data {
  repeated bytes Txs = 1 [packed = true];
}
```

Reply:
:(, protobuf why you no treat bytes as primitives :(

Good to know though, thank you!